external/fv3fit/fv3fit/reservoir/transformers/transformer.py

```diff
@@ -20,3 +27,85 @@ def encode(self, x: Sequence[ArrayLike]) -> np.ndarray:
     @abc.abstractmethod
     def decode(self, x: ArrayLike) -> Sequence[ArrayLike]:
         pass
+
+
+@io.register("do-nothing-transformer")
+class DoNothingAutoencoder(Transformer, Reloadable):
+    _CONFIG_NAME = "mock_transformer.yaml"
+
+    """Useful class for tests. Encode just concatenates input
+    variables. Decode separates them back into individual arrays.
+    """
+
+    def __init__(
+        self, latent_dim_len, original_feature_sizes: Optional[Sequence[int]] = None
+    ):
+        self._latent_dim_len = latent_dim_len
+        self.original_feature_sizes = original_feature_sizes
+
+    @property
+    def n_latent_dims(self):
+        return self._latent_dim_len
+
+    def encode(self, x):
+        self.original_feature_sizes = [arr.shape[-1] for arr in x]
+        return np.concatenate(x, -1)
+
+    def decode(self, latent_x):
+        if self.original_feature_sizes is None:
+            raise ValueError("Must encode data before decoding.")
+
+        split_indices = np.cumsum(self.original_feature_sizes)[:-1]
+        return np.split(latent_x, split_indices, axis=-1)
+
+    def dump(self, path: str) -> None:
+        with fsspec.open(os.path.join(path, self._CONFIG_NAME), "w") as f:
+            yaml.dump(
+                {
+                    "latent_dim_len": self.n_latent_dims,
+                    "original_feature_sizes": self.original_feature_sizes,
+                },
+                f,
+            )
+
+    @classmethod
+    def load(cls, path: str) -> "DoNothingAutoencoder":
+        with fsspec.open(os.path.join(path, cls._CONFIG_NAME), "r") as f:
+            config = yaml.safe_load(f)
+        return cls(
+            latent_dim_len=config["latent_dim_len"],
+            original_feature_sizes=config["original_feature_sizes"],
+        )
+
+
+def decode_columns(
+    encoded_output: np.ndarray, transformer: Transformer, xy_shape: Sequence[int]
+) -> Sequence[np.ndarray]:
+    # Differs from encode_columns as the decoder expects a single input array
```

Comment:
Please use docstrings for encode/decode instead of a comment, so this description will show up in an IDE.